src/rust/engine/src/externs/options.rs

```diff
@@ -146,11 +161,121 @@ impl PyConfigSource {
 #[pyclass]
 struct PyOptionParser(OptionParser);
 
-type RankedVal<T> = (T, isize);
```

Comment:
I didn't like this name any more, as once we add derivation info this is no longer compatible with the Python-side type of the same name. So this is now `OptionValue`.